lib/collection/src/shards/shard_holder.rs

```diff
@@ -323,6 +361,37 @@ impl ShardHolder {
         Ok(())
     }
 
+    pub fn resharding_filter(&self) -> Option<Filter> {
+        let state = self.resharding_state.read();
+
+        let Some(state) = state.deref() else {
+            return None;
+        };
+
+        if !state.filter_read_operations {
+            return None;
+        }
+
+        let Some(ring) = self.rings.get(&state.shard_key) else {
+            return None; // TODO(resharding)!?
+        };
+
+        let HashRing::Resharding { new, .. } = ring else {
+            return None; // TODO(resharding)!?
+        };
+
+        let filter = {
+            let ring = new.clone();
```

Comment:
I cannot imagine how this would have been without `Clone` being there :face_holding_back_tears:

Reply:
Annoying af. 😂 Thanks for pushing `Clone` into upstream!